pkg/kn/root/root.go

```diff
@@ -77,28 +70,47 @@ func NewRootCommand() (*cobra.Command, error) {
 	rootCmd.PersistentFlags().StringVar(&p.KubeCfgPath, "kubeconfig", "", "kubectl configuration file (default: ~/.kube/config)")
 	flags.AddBothBoolFlags(rootCmd.PersistentFlags(), &p.LogHTTP, "log-http", "", false, "log http traffic")
 
-	// root child commands
-	rootCmd.AddCommand(service.NewServiceCommand(p))
-	rootCmd.AddCommand(revision.NewRevisionCommand(p))
-	rootCmd.AddCommand(plugin.NewPluginCommand(p))
-	rootCmd.AddCommand(route.NewRouteCommand(p))
-	rootCmd.AddCommand(completion.NewCompletionCommand(p))
-	rootCmd.AddCommand(version.NewVersionCommand(p))
-	rootCmd.AddCommand(source.NewSourceCommand(p))
-	rootCmd.AddCommand(trigger.NewTriggerCommand(p))
+	// Grouped commands
+	groups := templates.CommandGroups{
+		{
+			Header: "Serving Commands:",
+			Commands: []*cobra.Command{
+				service.NewServiceCommand(p),
+				revision.NewRevisionCommand(p),
+				route.NewRouteCommand(p),
+			},
+		},
+		{
+			Header: "Eventing Commands:",
+			Commands: []*cobra.Command{
+				source.NewSourceCommand(p),
+				trigger.NewTriggerCommand(p),
+			},
+		},
+		{
+			Header: "Other Commands:",
+			Commands: []*cobra.Command{
+				plugin.NewPluginCommand(p),
+				completion.NewCompletionCommand(p),
+				version.NewVersionCommand(p),
```

Comment:
@rhuss : should this include `options` command as well?

Reply:
Kubectl doesn't include "options" as a regular command, so I tend to do the same. It's state prominently at the bottom how to get to the global options. And "options" is really kind of a no-op, much like "help" which is also not listed. So I think, that's ok to not include it here.